### PR TITLE
Revise security and privacy considerations

### PR DESCRIPTION
**Pull request by bifurcation**

### CODE REVIEW

**Review by raphaelrobert — APPROVED** [review]

File: draft-ietf-mls-architecture.md [hunk 1]
```diff
@@ -698,12 +698,12 @@ assumes that the attacker has complete control of the network. It is
...
 
--- The attacker can monitor the entire network.
+- The attacker can monitor the entire network.
 
--- The attacker can read unprotected messages.
+- The attacker can read unprotected messages.
 
--- The attacker can generate and inject any message in the unprotected
-   transport layer.
+- The attacker can generate and inject any message in the unprotected
```
Comment (kkohbrok) [inline draft-ietf-mls-architecture.md:705]:
```suggestion
- The attacker can generate, inject and delete any message in the unprotected
```

File: draft-ietf-mls-architecture.md [hunk 2]
```diff
@@ -795,18 +795,24 @@ messages to the Delivery Service or the MLS clients.
...
-[[TODO: It is unclear from this text whether MLS is "sensitive" in that it
-provides additional constraints to prevent this, or is "sensitive" in that it is
-vulnerable. Need to clarify]]
-
-The Delivery Service can have the role of helping with reliability,
-but is mainly useful for reliability in the asynchronous aspect of the
-communication between MLS clients.
+As noted above, MLS is designed to provide some robustness in the face of
+tampering within the secure transport, i.e., tampering by the Delivery Service.
+The confidentiality and authenticity properties of MLS prevent the DS reading or
```
Comment (kkohbrok) [inline draft-ietf-mls-architecture.md:800]: This is suggesting that the DS is a service running on a server outside of the client, where with #87, the definition would shift towards the DS being more of an abstract layer on top of an MLS client.

File: draft-ietf-mls-architecture.md [hunk 3]
```diff
@@ -795,18 +795,24 @@ messages to the Delivery Service or the MLS clients.
...
+writing messages.  MLS also provides a few tools for detecting message
+suppression, with the caveat that message suppression cannot always be
+distinguished from transport failure.
+
+Each encrypted MLS message carries a "generation" number which is per-sender
```
Comment (kkohbrok) [inline draft-ietf-mls-architecture.md:805]:
```suggestion
Each encrypted MLS message carries a "generation" number which is a per-sender
```

File: draft-ietf-mls-architecture.md [hunk 4]
```diff
@@ -795,18 +795,24 @@ messages to the Delivery Service or the MLS clients.
...
+incrementing counter.  If a group member observes a gap in the generation
+sequence for a sender, then they know that they have missed a message from that
+sender.  MLS also provides a facility for group members to send authenticated
+acknowledgements of application messages received within a group.
+
+As discusssed in {{delivery-service}}, the Delivery Service is trusted to select
+the single Commit message that is applied in each epoch from among the ones sent
+by group members.  Since only one Commit per epoch is meaningful, it's not
+useful for the DS to transmit multiple Commits too clients.  The risk remains
```
Comment (kkohbrok) [inline draft-ietf-mls-architecture.md:814]:
```suggestion
useful for the DS to transmit multiple Commits to clients.  The risk remains
```

File: draft-ietf-mls-architecture.md [hunk 5]
```diff
@@ -795,18 +795,24 @@ messages to the Delivery Service or the MLS clients.
...
+that the DS will use the ability maiciously.
```
Comment (kkohbrok) [inline draft-ietf-mls-architecture.md:815]:
```suggestion
that the DS will use the ability maliciously.
```

File: draft-ietf-mls-architecture.md [hunk 6]
```diff
@@ -894,8 +900,14 @@ Additionally, some services require that a recipient be able to prove to the
 service provider that a message was sent by a given client, in order to report
 abuse. MLS supports both of these use cases. In some deployments, these services
 are provided by mechanisms which allow the receiver to prove a message's origin
-to a third party (this if often called "non-repudiation"), but it should also be
-possible to operate MLS in a "deniable" mode where such proof is not possible.
+to a third party. This is often called "non-repudiation". 
+
+Roughly speaking, "deniability" is the opposite of "non-repudiation", i.e., the
+property that it is impossible to prove to a third party that a message was sent
```
Comment (kkohbrok) [inline draft-ietf-mls-architecture.md:906]: The meaning is clear here, but "third party" is a bit odd in a group-messaging setting. Maybe (n+1)th party? 🤔

File: draft-ietf-mls-architecture.md [hunk 7]
```diff
@@ -943,7 +955,7 @@ message to a group which appears to be from any valid client since
 they cannot forge the signature.
 
 The MLS protocol will ensure that an adversary cannot compute any
-previous AEAD keys for the same epoch, or any other epochs.  Because
+previous or future AEAD keys for the same epoch, or any other epochs.  Because
```
Comment (kkohbrok) [inline draft-ietf-mls-architecture.md:958]: I'm not sure this is correct. After a full state compromise, the adversary can compute future AEAD keys for that epoch. We're not getting intra-epoch PCS with MLS. To make that work, we'd have to KEM an additional secret to the other group members with every application message. But that would make out-of-order messages impossible and application messages generally less efficient.

**Review by kkohbrok — CHANGES_REQUESTED** [review]
Thanks for cleaning up a bit! Looks good in general, just a few nits.

File: draft-ietf-mls-architecture.md (hunk 6 above)
Comment (beurdouche) [reply draft-ietf-mls-architecture.md:906]: I didn't see a much better way to phrase this, so I kept third party...

File: draft-ietf-mls-architecture.md (hunk 7 above)
Comment (beurdouche) [reply draft-ietf-mls-architecture.md:958]: I agree, will undo that change.

**Pull request merged**